core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -54,22 +56,24 @@
   private final TableOperations ops;
   private final Table table;
   private final Long snapshotId;
+  private final Map<String, String> options;
   private final Schema schema;
   private final Expression rowFilter;
   private final boolean caseSensitive;
   private final boolean colStats;
   private final Collection<String> selectedColumns;
 
   protected BaseTableScan(TableOperations ops, Table table, Schema schema) {
-    this(ops, table, null, schema, Expressions.alwaysTrue(), true, false, null);
+    this(ops, table, null, null, schema, Expressions.alwaysTrue(), true, false, null);
```

Comment:
You can avoid null checks by always having a map instance. Using `ImmutableMap.of()` would be easier.

Reply:
Using an `ImmutableMap` all around now.